include/grpcpp/grpc_library.h

```diff
@@ -16,13 +16,24 @@
  *
  */
 
-// DEPRECATED: The headers in include/grpc++ are deprecated. Please include the
-// headers in include/grpcpp instead. This header exists only for backwards
-// compatibility.
+#ifndef GRPCPP_GRPC_LIBRARY_H
+#define GRPCPP_GRPC_LIBRARY_H
 
-#ifndef GRPCXX_IMPL_CODEGEN_GRPC_LIBRARY_H
-#define GRPCXX_IMPL_CODEGEN_GRPC_LIBRARY_H
+#include <grpcpp/impl/grpc_library.h>
 
-#include <grpcpp/impl/codegen/grpc_library.h>
+namespace grpc {
 
-#endif  // GRPCXX_IMPL_CODEGEN_GRPC_LIBRARY_H
+/// Classes that require gRPC to be initialized should inherit from this class.
+class GrpcLibrary {
+ public:
+  explicit GrpcLibrary(bool call_grpc_init = true);
+  virtual ~GrpcLibrary();
+
+ private:
+  bool grpc_init_called_;
+  static internal::GrpcLibrary internal_grpc_library;
```

Comment:
Why is there a static grpc::internal::GrpcLibrary ? It seems like its methods only call grpc_init() and grpc_shutdown(). Why not call them directly in grpc::GrpcLibrary constructor and destructor respectively ?